### ISSUE TITLE
mDNS broken with AppArmor denials for AF_NETLINK since update to Qt 5.14

### ISSUE THREAD

**Issue by cgutman** [maxiberta/moonlight-snap#9]
Since the edge snap updated to Qt 5.14, I'm seeing the log full of AF_NETLINK errors (from Qt itself, since Moonlight never creates netlink sockets). It also seems to be correlated with loss of automatic mDNS auto-discovery functionality.

The messages also repeat every 10 seconds which happens to be the interval that QMdnsEngine sends mDNS queries, so that further indicates something mDNS-related is going wrong. I'm not seeing any of Moonlight's expected outbound mDNS queries in Wireshark.

I tried to `snap run --gdb moonlight` to determine the source of the messages but Qt failed to load the `xcb` platform plugin and wouldn't initialize under GDB.

Interestingly, I noticed this problem both with core20's native Qt and core18+Neon PPA.

Installing with `--devmode` fixes both the AF_NETLINK denials and mDNS autodiscovery.

Stdout from moonlight:
```
00:07:00 - Qt Critical: Could not create AF_NETLINK socket (Permission denied)
00:07:00 - Qt Critical: Could not create AF_NETLINK socket (Permission denied)
00:07:00 - Qt Critical: Could not create AF_NETLINK socket (Permission denied)
00:07:00 - Qt Critical: Could not create AF_NETLINK socket (Permission denied)
00:07:00 - Qt Critical: Could not create AF_NETLINK socket (Permission denied)
00:07:10 - Qt Critical: Could not create AF_NETLINK socket (Permission denied)
00:07:10 - Qt Critical: Could not create AF_NETLINK socket (Permission denied)
00:07:10 - Qt Critical: Could not create AF_NETLINK socket (Permission denied)
00:07:10 - Qt Critical: Could not create AF_NETLINK socket (Permission denied)
```

The following message is spammed over and over `dmesg`:
```
[43929.065137] audit: type=1400 audit(1589559656.939:5446): apparmor="DENIED" operation="create" profile="snap.moonlight.moonlight" pid=27398 comm="moonlight" family="netlink" sock_type="dgram" protocol=0 requested_mask="create" denied_mask="create"
``

**Comment by maxiberta** [comment]
Hm, interesting. Either rebuilding with Qt 5.9, installing with `--devmode`, or manually connecting any of `network-observe` or `netlink-connector` fix both the mDNS discovery and the apparmor denials (while `avahi-observe` makes no difference).

Posted [a question in the forum](https://forum.snapcraft.io/t/moonlight-mdns-broken-with-apparmor-denials-for-af-netlink-since-update-to-qt-5-14/17549).

**Comment by maxiberta** [comment]
As per [the forum discussion](https://forum.snapcraft.io/t/auto-connection-request-for-moonlight-network-observe/17576), `netlink dgram` will be added to the `network` interface in the next `snapd` release, fixing our issue. In the meantime, this can be fixed by manually connecting:

```
$ sudo snap connect moonlight:network-observe
```

Thanks for your feedback @cgutman!

**Comment by cgutman** [comment]
I confirmed locally this is fixed in snapd 2.45.1 (beta channel)
